src/libs/actions/OnyxDerived/configs/reportTransactionsAndViolations.ts

```diff
@@ -0,0 +1,70 @@
+import type {OnyxCollection} from 'react-native-onyx';
+import createOnyxDerivedValueConfig from '@userActions/OnyxDerived/createOnyxDerivedValueConfig';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Transaction} from '@src/types/onyx';
+
+let previousTransactions: OnyxCollection<Transaction> = {};
+
+export default createOnyxDerivedValueConfig({
+    key: ONYXKEYS.DERIVED.REPORT_TRANSACTIONS_AND_VIOLATIONS,
+    dependencies: [ONYXKEYS.COLLECTION.TRANSACTION, ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS],
+    compute: ([transactions, violations], {sourceValues, currentValue}) => {
+        if (!transactions) {
+            return {};
+        }
+
+        const transactionsUpdates = sourceValues?.[ONYXKEYS.COLLECTION.TRANSACTION];
+        const transactionViolationsUpdates = sourceValues?.[ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS];
+        let transactionsToProcess = Object.keys(transactions);
+        if (transactionsUpdates) {
+            transactionsToProcess = Object.keys(transactionsUpdates);
+        } else if (transactionViolationsUpdates) {
+            transactionsToProcess = Object.keys(transactionViolationsUpdates).map((transactionViolation) =>
+                transactionViolation.replace(ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS, ONYXKEYS.COLLECTION.TRANSACTION),
+            );
+        }
+
+        const reportTransactionsAndViolations = currentValue ?? {};
+        for (const transactionKey of transactionsToProcess) {
+            const transaction = transactions[transactionKey];
+            const reportID = transaction?.reportID;
+
+            // If the reportID of the transaction has changed (e.g. the transaction was split into multiple reports), we need to delete the transaction from the previous reportID and the violations from the previous reportID
+            const previousTransaction = previousTransactions?.[transactionKey];
+            const previousReportID = previousTransaction?.reportID;
+
+            if (previousReportID && previousReportID !== reportID && reportTransactionsAndViolations[previousReportID]) {
+                delete reportTransactionsAndViolations[previousReportID].transactions[transactionKey];
+                const transactionID = previousTransaction?.transactionID;
+                if (transactionID) {
+                    delete reportTransactionsAndViolations[previousReportID].violations[`${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`];
+                }
+            }
+
+            if (!reportID) {
+                // eslint-disable-next-line no-continue
```

Comment:
We should remove this ESLint config, it's outdated imo since we allow/encourage regular for...of loops now.

Reply:
@roryabraham yeah I agree, could you create an issue for this and assign me? There is a couple of more places in the codebase that are still using so I could remove them all in a single PR

Reply:
Created https://github.com/Expensify/App/issues/65762